README.md

```diff
@@ -1,98 +1,120 @@
+- [**mongodb-ace-theme**](packages/ace-theme): MongoDB Theme for the ACE Editor
+- [**mongodb-ace-theme-query**](packages/ace-theme-query): MongoDB Theme for the ACE Editor in the Query Bar
+- [**mongodb-collection-model**](packages/collection-model): MongoDB collection model.
+- [**mongodb-connection-model**](packages/connection-model): MongoDB connection model.
+- [**mongodb-data-service**](packages/data-service): MongoDB Data Service
+- [**mongodb-database-model**](packages/database-model): MongoDB database model.
+- [**mongodb-explain-compat**](packages/mongodb-explain-compat): Convert mongodb SBE explain output to 4.4 explain output
+- [**mongodb-explain-plan-model**](packages/explain-plan-model): Ampersand model abstraction for MongoDB explain plans (3.0+)
+- [**mongodb-index-model**](packages/index-model): MongoDB index model.
+- [**mongodb-instance-model**](packages/instance-model): MongoDB instance model.
+- [**mongodb-js-metrics**](packages/metrics): Shareable metric recording.
+- [**mongodb-language-model**](packages/mongodb-language-model): Parses MongoDB query language and returns an abstract syntax tree
+- [**mongodb-security**](packages/security): Portable business logic of MongoDB security model
+- [**storage-mixin**](packages/storage-mixin): Ampersand model mixin to persist data via various storage backends
+
+## Working With the Monorepo
+
+You'll need node ^12.4.0 and npm 7 installed on your machine to work with the repository locally. After your environment is ready, navigate to the repository and run `npm run bootstrap`, this will install dependencies and will compile all packages.
+
+After bootstrap is finished, you should be able to run `npm run start` and see Compass application running locally.
+
+This monorepo is powered by [`npm workspaces`](https://docs.npmjs.com/cli/v7/using-npm/workspaces) an [`lerna`](https://github.com/lerna/lerna#readme), although not necessary, it might be helpful to have a high level understanding of those tools.
+
+### Working on Plugins
+
+Most of the plugins have their own development environment so you can work on them in isolation. If you want to work on plugin without running the whole Compass application, you can run `npm run start` in the plugin directory, either with the help of `lerna` or `npm workspaces`. For example to start compass-connect plugin locally you can run `npm run start --workspace @mongodb-js/compass-connect` or `npx lerna run start --scope @mongodb-js/compass-connect --stream`.
+
+If you want to see your changes applied in Compass, you might need to rebuild plugins that you changed with the `compile` command. Instead of manually writing out the `scope` you might want to use `lerna --since` filter to rebuild everything since your local or origin `HEAD` of the git history: `npx lerna run compile --stream --since origin/HEAD`. Restarting or hard-reloading (Shift+CMD+R) Compass after compilation is finished should apply your changes.
```

Comment:
Would be good to have this as `npm run compile-changed` probably:
`npx lerna run compile --stream --since origin/HEAD`

Reply:
How does ebb46ab look?